pkg/sessionctx/variable/sysvar.go

```diff
@@ -2779,6 +2779,15 @@ var defaultSysVars = []*SysVar{
 	}, GetGlobal: func(ctx context.Context, vars *SessionVars) (string, error) {
 		return BoolToOnOff(EnableResourceControl.Load()), nil
 	}},
+	{Scope: ScopeGlobal, Name: TiDBResourceControlStrictMode, Value: BoolToOnOff(DefTiDBResourceControlStrictMode), Type: TypeBool, SetGlobal: func(ctx context.Context, vars *SessionVars, s string) error {
+		if TiDBOptOn(s) != EnableResourceControl.Load() {
```

Comment:
```suggestion
		if TiDBOptOn(s) != EnableResourceControlStrictMode.Load() {
```
should be `EnableResourceControlStrictMode`?

Reply:
fixed. It's weird that unit test does not find this bug.